src/RandMsg.chpl

```diff
@@ -29,137 +32,53 @@ module RandMsg
 
     :arg reqMsg: message to process (contains cmd,aMin,aMax,len,dtype)
     */
-    @arkouda.registerND
-    proc randintMsg(cmd: string, msgArgs: borrowed MessageArgs, st: borrowed SymTab, param nd: int): MsgTuple throws {
-        param pn = Reflection.getRoutineName();
-        var repMsg: string; // response message
-        
-        const shape = msgArgs.get("shape").getTuple(nd);
-        const dtype = str2dtype(msgArgs.getValueOf("dtype"));
-        const seed = msgArgs.getValueOf("seed");
-        const low = msgArgs.get("low");
-        const high = msgArgs.get("high");
+    @arkouda.instantiateAndRegister
+    proc randint(cmd: string, msgArgs: borrowed MessageArgs, st: borrowed SymTab, type array_dtype, param array_nd: int): MsgTuple throws
+        where array_dtype != BigInteger.bigint
+    {
+        const shape = msgArgs["shape"].toScalarTuple(int, array_nd),
+              seed = msgArgs["seed"].toScalar(int);
+
+        const low = msgArgs["low"].toScalar(array_dtype),
+              high = msgArgs["high"].toScalar(array_dtype) - if isIntegralType(array_dtype) then 1 else 0;
 
         var len = 1;
         for s in shape do len *= s;
+        overMemLimit(len);
 
-        // get next symbol name
-        const rname = st.nextName();
-
-        // if verbose print action
         randLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),
-               "cmd: %s len: %i dtype: %s rname: %s aMin: %s: aMax: %s".format(
-                                           cmd,len,dtype2str(dtype),rname,low.getValue(),high.getValue()));
-
-        proc doFillRand(type t, param sub: t): MsgTuple throws {
-            overMemLimit(len);
-            const aMin = low.getScalarValue(t),
-                  aMax = high.getScalarValue(t) - sub,
-                  t1 = Time.timeSinceEpoch().totalSeconds();
-            var e = st.addEntry(rname, (...shape), t);
-            randLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),
-                              "alloc time = %i sec".format(Time.timeSinceEpoch().totalSeconds() - t1));
-            const t2 = Time.timeSinceEpoch().totalSeconds();
-            fillRand(e.a, aMin, aMax, seed);
-            randLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),
-                              "compute time = %i sec".format(Time.timeSinceEpoch().totalSeconds() - t2));
+                         "cmd: %s len: %i dtype: %s aMin: %?: aMax: %?".format(
+                         cmd,len,type2str(array_dtype),low,high));
 
-            const repMsg = "created " + st.attrib(rname);
-            randLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),repMsg);
-            return new MsgTuple(repMsg, MsgType.NORMAL);
-        }
+        var t = new stopwatch();
+        t.start();
```

Comment:
first time seeing the stopwatch. I've still been using `timeSinceEpoch().totalSeconds()`, I'll have to remember this

Reply:
in fact I'll go update the chpl tutorial to use it now!